docs/source/devel/index.rst

```diff
@@ -94,66 +94,43 @@ We use the 'waiting time' to alternate between tasks.
    and `remove_targets
    <repository_service_tuf_worker.html#repository_service_tuf_worker.repository.MetadataRepository.remove_targets>`_
 
-Before the Repository Worker adds/removes the new target and does steps 1 to 3,
-it Locks [#f1]_ specific Hashed Bins Role, for example, bins-a.
+Repository Worker adds/removes the target to the SQL Database.
 
+It means the multiple Repository Workers can write multiple Targets
+(``TargetFiles``) simultaneously from various tasks in the Database.
 
-It means the multiple Repository Workers can write multiple Hashed Bins Roles
-simultaneously from various tasks.
+When a task finishes, it send a task the ``publish_targets``.
 
-When Step 3 finishes, the Repository Worker will lock  ``unpublished_meta`` and
-add the Hashed Bins Role in the list if it does not exist.
+Every minute, the routine task **Publish the new Hashed Bins Roles** also runs.
 
-.. uml::
-
-  @startuml
-      !pragma useVerticalIf
-      start
-      partition "add/remove targets" {
-         :Group targets per Hashed Bins Role;
-         repeat
-         repeat while (Try Lock Hashed Bins Roles) is (Waiting)
-            :Lock Hashed Bins;
-            :Add the target(s) to the Hashed Bin Role;
-            :Generate a new version;
-            :Persist the new Hashed Bin Role in the Storage;
-            repeat
-            repeat while (Try Lock unpublished_meta) is (Waiting)
-            :Lock unpublished_meta;
-            if (Hashed Bins Role in umpublished_meta) then (not in)
-               :Add Hashed Bins name;
-            endif
-            :Unlock unpublished_meta;
-            :Unlock Hashed Bins Roles;
-         repeat
-         repeat while (all Hashed Bins are Published) is (Waiting)
-         stop
-      }
-   @enduml
+The task will continue run, wait until all the targets are persisted to the
+Repository Metadata backend.
 
-Every minute, the routine task **Publish the new Hashed Bins Roles** runs and
-gets the names of the unpublished Hashed Bins Roles, looks in the Storage the
-latest version and runs steps 4 to 9, and flushes the ``unpublished_meta``.
+The **Publish the new Hashed Bins Roles** task (``publish_targets``) runs once
+per time to using locks [#f1]_ . It will  will do:
```

Comment:
Could you add a sentence or two about how we synchronize the state of the database with the state of the JSON metadata files?

Reply:
@MVrachev  Added the mention.